### PR TITLE
Fix golint error for staging/src/k8s.io/client-go/discovery/fake

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fix golint errors from `staging/src/k8s.io/client-go/discovery/fake` and remove the directory from `hack/.golint_failures`.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
Errors from golint:

> staging/src/k8s.io/client-go/discovery/fake/discovery.go:34:6: type name will be used as fake.FakeDiscovery by other packages, and that stutters; consider calling this Discovery

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

### CODE REVIEW

**Comment by RainbowMango** [comment]
It's my first time to deal with generated files.
I tried running `make update`, but the error happens when update docs.

> Running update-generated-docs
> # k8s.io/kubernetes/vendor/k8s.io/client-go/kubernetes/fake
> vendor/k8s.io/client-go/kubernetes/fake/clientset_generated.go:137:13: undefined:  "k8s.io/kubernetes/vendor/k8s.io/client-go/discovery/fake".FakeDiscovery
> !!! [0814 10:24:43] Call tree:
> !!! [0814 10:24:43]  1: /root/go/src/k8s.io/kubernetes/hack/lib/golang.sh:714  kube::golang::build_some_binaries(...)
> !!! [0814 10:24:43]  2: /root/go/src/k8s.io/kubernetes/hack/lib/golang.sh:853 kube::golang::build_binaries_for_platform(...)
> !!! [0814 10:24:43]  3: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
> !!! [0814 10:24:43] Call tree:
> !!! [0814 10:24:43]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
> !!! [0814 10:24:43] Call tree:
> !!! [0814 10:24:43]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
> make[1]: *** [all] Error 1
> Running update-generated-docs FAILED
> make: *** [update] Error 1

**Comment by RainbowMango** [comment]
Seems fake. Let's test again.
`W0814 12:39:22.231] Server terminated abruptly (error code: 14, error message: '', log file: '/bazel-scratch/.cache/bazel/_bazel_prow/48d5366022b4e3197674c8d6e2bee219/server/jvm.out')`

/retest

File: staging/src/k8s.io/client-go/discovery/fake/discovery.go [hunk 1]
```diff
@@ -29,16 +29,16 @@ import (
 	"k8s.io/client-go/testing"
 )
 
-// FakeDiscovery implements discovery.DiscoveryInterface and sometimes calls testing.Fake.Invoke with an action,
+// Discovery implements discovery.DiscoveryInterface and sometimes calls testing.Fake.Invoke with an action,
 // but doesn't respect the return value if any. There is a way to fake static values like ServerVersion by using the Faked... fields on the struct.
-type FakeDiscovery struct {
+type Discovery struct {
```
Comment (sttts) [inline staging/src/k8s.io/client-go/discovery/fake/discovery.go]: we cannot change this. It breaks tons of 3rdparty code.

Comment (RainbowMango) [reply staging/src/k8s.io/client-go/discovery/fake/discovery.go]: OK, thanks.  I will close this PR.

**Pull request closed without merging**